tonic-web/src/call.rs

```diff
@@ -395,7 +395,7 @@ fn decode_trailers_frame(mut buf: Bytes) -> Result<Option<HeaderMap>, Status> {
     for (i, b) in buf.iter().enumerate() {
         if b == &b'\r' && buf.get(i + 1) == Some(&b'\n') {
             let trailer = temp_buf.copy_to_bytes(i - cursor_pos);
-            cursor_pos = i;
+            cursor_pos = i + 2;
```

Comment:
Could you please add a few comments around this loop that explain the conceptual logic here?

Reply:
yea I agree, I'm not entirely convinced this is the most idiomatic rust to parse the trailers, but I wanted to do the minimal change. Comments added to my understanding of what is going on, PTAL

Reply:
Thanks. Trying to reason about this code hurts my head, but I suppose this is an improvement at least. Will probably try to rewrite this.